File: Sheet 3 (Loops 1)/Y.cpp
#include <iostream>
#include <vector>
#include <queue>
#include <deque>
#include <algorithm>
using namespace std;

int main()
{
    ios_base::sync_with_stdio(0) ;
    cin.tie(0) ;
    int n , c = 0 ,maxi = 0;
    string x;
    cin >> n >> x;
    for (int i =0 ;i < x.size() ; i ++){
        if (x[i]=='D')c--;
        else c++;
        maxi = max(maxi , c);
    }
    cout << maxi ;


    return 0;
}

File: Sheet 3 (Loops 1)/Y.py
n = int(input())
a = input()
c = 0 ; maxi = 0
for i in range(len(a)):
    if a[i] == 'I':
        c+=1
    else:
        c-=1
    maxi = max(maxi , c)
print(maxi)
